### PR TITLE
CB-19105 Throw CloudImageException on marketplace related provisioning errors

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java [hunk 1]
```diff
@@ -84,6 +85,10 @@ public class AzureUtils {
 
     private static final int MAX_DISK_ENCRYPTION_SET_NAME_LENGTH = 80;
 
+    private static final String AUTHORIZATION_FAILED_CODE = "AuthorizationFailed";
```
Comment (pdarvasi) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java:88]: are you sure, is this _always_ related to Marketplace image errors?

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java [hunk 2]
```diff
@@ -760,6 +765,22 @@ public CloudConnectorException convertToCloudConnectorException(CloudException e
         }
     }
 
+    public CloudConnectorException convertToCloudException(CloudException e, String actionDescription) {
+        LOGGER.warn(String.format("%s failed, cloud exception happened:", actionDescription), e);
+        if (e.body() != null && e.body().details() != null) {
+            String details = e.body().details().stream().map(this::getCloudErrorMessage).collect(Collectors.joining(", "));
+            if (e.body().code().contains(AUTHORIZATION_FAILED_CODE) && e.body().message().contains(MICROSOFT_MARKETPLACE_ROLE)) {
+                return new CloudImageException(String.format("%s failed, status code %s, error message: %s, details: %s",
+                        actionDescription, e.body().code(), e.body().message(), details));
```
Comment (pdarvasi) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java:774]: could you pls use a variable for `e.body().message()` and `e.body().code()`?

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java (hunk 1 above)
Comment (daszabo) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java:88]: Absolutley not, but combined with the Marketplace role in the message, it should be.

**Review by pdarvasi — APPROVED** [review]

**Comment by daszabo** [comment]
retest this please

**Comment by daszabo** [comment]
retest this please

**Review by foldik — APPROVED** [review]
LGTM
